### ISSUE TITLE
Unable to Telemetry  Stream mdt-out to telegraf collector (Cisco-IOS-XR-qos-ma-oper:qos/interface-table/interface/input/service-policy-names/service-policy-instance/statistics)

### ISSUE THREAD

**Issue by anpedro** [influxdata/telegraf#12635]
### Relevant telegraf.conf

```toml
#[global_tags]
#  namespace = "lab"

[agent]
  interval = "15s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 10000
 # metric_batch_size = 60000
 # metric_buffer_limit = 100000
  collection_jitter = "0s"
  flush_interval = "100s"
  flush_jitter = "0s"
  precision = ""
  debug = true
  quiet = false
  logfile = ""
  hostname = "telegraf-1"
  omit_hostname = true

[[inputs.cisco_telemetry_mdt]]
  transport = "tcp"
  service_address = ":6666"
#  embedded_tags = [
#    "Cisco-IOS-XR-platforms-ofa-oper:ofa/stats/nodes/node/xr-8000-platforms-npu-resources-oper:hw-resources-datas/xr-8000-platforms-npu-resources-oper:hw-resources-data/xr-8000-platforms-npu-resources-oper:npu-hwr/xr-8000-platforms-npu-resources-oper:bank/xr-8000-platforms-npu-resources-oper:oor-state/xr-8000-platforms-npu-resources-oper:oor-state"
#  ]

[[processors.converter]]
  namepass = ["Cisco-IOS-XR-procmem-oper:processes-memory/nodes/node/process-ids/process-id"]
  #order = 1
  [processors.converter.fields]
    tag = ["name"]


#[[processors.rename]]
#  namepass = ["Cisco-IOS-XR-procmem-oper:processes-memory/nodes/node/process-ids/process-id"]
#  order = 2
#  [[processors.rename.replace]]
#    tag = "name"
#    dest = "process_name"

#[[processors.converter]]
#  namepass = ["Cisco-IOS-XR-platforms-ofa-oper:ofa/stats/nodes/node/Cisco-IOS-XR-8000-platforms-npu-resources-oper:hw-resources-datas/hw-resources-data"]
#  [processors.converter.fields]
#    integer = ["npu_hwr/bank/oor_state/oor_state"]


#[[processors.converter]]
#  namepass = ["Cisco-IOS-XR-platforms-ofa-oper:ofa/stats/nodes/node/Cisco-IOS-XR-8000-platforms-npu-resources-oper:hw-resources-datas/hw-resources-data"]
#  [processors.converter.fields]
#    integer = ["npu_hwr/npu_id"]


[[outputs.influxdb_v2]]
  urls = ["http://localhost:8086"]
  insecure_skip_verify = true
  content_encoding = "gzip"
  # InfluxDB's admin Token for simplicity
  token = "Qy-c3KkLOG84V_W9QMm867wr5bQ7BGy0Ab0yS0uMauHfsyFR7inEaCtxG_Jtoek9JbQST83H1gCVuecUSTja9A=="
  organization = "Meta"
  bucket = "cisco_mdt_dialout_ios_xr"

#[[outputs.file]]
#  files = ["stdout"]
#  data_format = "influx"

[[outputs.file]]
  files = ["stdout"]
  data_format = "json"
  json_timestamp_units = "1ms"

#[[outputs.file]]
#  files = ["stdout"]
#  data_format = "prometheus"

#[[outputs.prometheus_client]]
  ## Address to listen on.
#  listen = ":9273"
#  metric_version = 2

#[[outputs.health]]
#  service_address = "http://:8088"
#  namepass = ["internal_write"]
#  tagpass = { output = ["influxdb"] }
# [[outputs.health.compares]]
#    field = "buffer_size"
#    lt = 100.0
```

### Logs from Telegraf

```text
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]: 2023-02-04T22:37:43Z D! [inputs.cisco_telemetry_mdt] No measurement alias for encoding path: Cisco-IOS-XR-qos-ma-oper:qos/interface-table/interface/input/service-policy-names/service-policy-instance/statistics
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]: panic: runtime error: index out of range [1] with length 0
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]:
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]: goroutine 122 [running]:
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]: github.com/influxdata/telegraf/plugins/inputs/cisco_telemetry_mdt.(*CiscoTelemetryMDT).handleTelemetry(0xc000e42780, {0xc001100000, 0x10416, 0x20000})
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go:412 +0x8e7
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]: github.com/influxdata/telegraf/plugins/inputs/cisco_telemetry_mdt.(*CiscoTelemetryMDT).handleTCPClient(0xc000e42780, {0x6cab1c0?, 0xc000724660})
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go:305 +0x8f
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]: github.com/influxdata/telegraf/plugins/inputs/cisco_telemetry_mdt.(*CiscoTelemetryMDT).acceptTCPClients.func1()
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go:240 +0xe9
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]: created by github.com/influxdata/telegraf/plugins/inputs/cisco_telemetry_mdt.(*CiscoTelemetryMDT).acceptTCPClients
Feb 04 17:37:43 server-podman-1 telegraf-mdt-dialout[20424]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go:238 +0x22a
Feb 04 17:37:43 server-podman-1 podman[20470]: 2023-02-04 17:37:43.65013652 -0500 EST m=+0.041736456 container died 3fd86b261e91b60e9d590b41f795c9155fd9a2703c911971e7ada90348d453e4 (image=docker.io/library/telegraf:1.25.1-alpine, name=telegraf-mdt-dialout, health_status=)
Feb 04 17:37:47 server-podman-1 podman[20470]: 2023-02-04 17:37:47.669000042 -0500 EST m=+4.060599924 container remove 3fd86b261e91b60e9d590b41f795c9155fd9a2703c911971e7ada90348d453e4 (image=docker.io/library/telegraf:1.25.1-alpine, name=telegraf-mdt-dialout, health_status=, PODMAN_SYSTEMD_UNIT=container-telegraf-mdt-dialout.service)
Feb 04 17:37:47 server-podman-1 systemd[1]: container-telegraf-mdt-dialout.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Feb 04 17:37:47 server-podman-1 systemd[1]: container-telegraf-mdt-dialout.service: Failed with result ‘exit-code’.
Feb 04 17:37:48 server-podman-1 systemd[1]: container-telegraf-mdt-dialout.service: Scheduled restart job, restart counter is at 13.
Feb 04 17:37:48 server-podman-1 systemd[1]: Stopped Podman container-telegraf-mdt-dialout.service.
Feb 04 17:37:48 server-podman-1 systemd[1]: Starting Podman container-telegraf-mdt-dialout.service...
Feb 04 17:37:48 server-podman-1 podman[20501]: 2023-02-04 17:37:48.078121144 -0500 EST m=+0.028260863 image pull  docker.io/telegraf:1.25.1-alpine
Feb 04 17:37:48 server-podman-1 podman[20501]:
Feb 04 17:37:48 server-podman-1 podman[20501]: 2023-02-04 17:37:48.72816325 -0500 EST m=+0.678303004 container create 42080aa419a690442380c1bb24fb56c2a51f84568c520f43455044f87b8734a6 (image=docker.io/library/telegraf:1.25.1-alpine, name=telegraf-mdt-dialout, health_status=, PODMAN_SYSTEMD_UNIT=container-telegraf-mdt-dialout.service)
Feb 04 17:37:48 server-podman-1 podman[20501]: 2023-02-04 17:37:48.835454949 -0500 EST m=+0.785594648 container init 42080aa419a690442380c1bb24fb56c2a51f84568c520f43455044f87b8734a6 (image=docker.io/library/telegraf:1.25.1-alpine, name=telegraf-mdt-dialout, health_status=, PODMAN_SYSTEMD_UNIT=container-telegraf-mdt-dialout.service)
Feb 04 17:37:48 server-podman-1 systemd[1]: Started Podman container-telegraf-mdt-dialout.service.
Feb 04 17:37:48 server-podman-1 podman[20501]: 2023-02-04 17:37:48.870594026 -0500 EST m=+0.820733730 container start 42080aa419a690442380c1bb24fb56c2a51f84568c520f43455044f87b8734a6 (image=docker.io/library/telegraf:1.25.1-alpine, name=telegraf-mdt-dialout, health_status=, PODMAN_SYSTEMD_UNIT=container-telegraf-mdt-dialout.service)
Feb 04 17:37:48 server-podman-1 podman[20501]: 42080aa419a690442380c1bb24fb56c2a51f84568c520f43455044f87b8734a6
```

### System info

Rocky Linux release 9.1 (Blue Onyx)

### Docker

the container is running on podman container.

4682de558220  docker.io/library/telegraf:1.25.1-alpine  telegraf              46 hours ago  Up 46 hours ago                      telegraf-mdt-dialout
[root@facebook-podman-1:/var/containers/telegraf]#

### Steps to reproduce

I can reproduce the issue just sending mdt-out data from IOS XR router from the given sensor-path:

Cisco-IOS-XR-qos-ma-oper:qos/interface-table/interface/input/service-policy-names/service-policy-instance/statistics

### Expected behavior

it is expected to have this data on the collector and send to influx for grafana queries.

### Actual behavior

telegraf podman container is crashing.

**Comment by anpedro** [comment]
hello, happy to do it!
how do I do that?

Can you provide step by step?

Thanks,

Andre Pedro

**Comment by powersj** [comment]
In 20-30mins there will be a comment added to the #12636 by the telegraf-tiger. The comment will include links to downloads, where you can grab an archive for your architecture. Download the binary you need.

Open the archive up, and run the telegraf binary with the telegraf config you are using.

See if you can reproduce the problem or if the issue is resolved.

Thanks!

**Comment by anpedro** [comment]
ok - let me know when the fix is available.

**Comment by anpedro** [comment]
I've downloaded the binary but it looks like I can't get the daemon to
start.

2023-02-07T12:46:04Z I! Starting Telegraf 1.26.0-b70711b4
2023-02-07T12:46:04Z I! Available plugins: 229 inputs, 9 aggregators, 26
processors, 21 parsers, 57 outputs, 2 secret-stores
2023-02-07T12:46:04Z I! Loaded inputs: cisco_telemetry_mdt (2x) cpu disk
diskio kernel mem processes swap system
2023-02-07T12:46:04Z I! Loaded aggregators:
2023-02-07T12:46:04Z I! Loaded processors: converter (2x)
2023-02-07T12:46:04Z I! Loaded secretstores:
2023-02-07T12:46:04Z I! Loaded outputs: file (2x) influxdb_v2 (2x)
2023-02-07T12:46:04Z I! Tags enabled: host=facebook-podman-1
2023-02-07T12:46:04Z I! [agent] Config: Interval:15s, Quiet:false,
Hostname:"telegraf-1", Flush Interval:1m40s
2023-02-07T12:46:04Z D! [agent] Initializing plugins
2023-02-07T12:46:04Z D! [agent] Connecting outputs
2023-02-07T12:46:04Z D! [agent] Attempting connection to
[outputs.influxdb_v2]
2023-02-07T12:46:04Z D! [agent] Successfully connected to
outputs.influxdb_v2
2023-02-07T12:46:04Z D! [agent] Attempting connection to [outputs.file]
2023-02-07T12:46:04Z D! [agent] Successfully connected to outputs.file
2023-02-07T12:46:04Z D! [agent] Attempting connection to [outputs.file]
2023-02-07T12:46:04Z D! [agent] Successfully connected to outputs.file
2023-02-07T12:46:04Z D! [agent] Attempting connection to
[outputs.influxdb_v2]
2023-02-07T12:46:04Z D! [agent] Successfully connected to
outputs.influxdb_v2
2023-02-07T12:46:04Z D! [agent] Starting service inputs
2023-02-07T12:46:04Z E! [telegraf] Error running agent: starting input
inputs.cisco_telemetry_mdt: listen tcp :6667: bind: address already in use



***@***.***:~/telegraf/telegraf-1.26.0/usr/bin]$ netstat -tupnl
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State
    PID/Program name
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN
     -
tcp6       0      0 :::9100                 :::*                    LISTEN
     -
tcp6       0      0 :::8080                 :::*                    LISTEN
     -
tcp6       0      0 :::22                   :::*                    LISTEN
     -
tcp6       0      0 :::3000                 :::*                    LISTEN
     -
udp        0      0 127.0.0.1:323           0.0.0.0:*
    -
udp6       0      0 ::1:323                 :::*
     -
udp6    2304      0 fe80::20c:29ff:fe7a:546 :::*
     -
***@***.***:~/telegraf/telegraf-1.26.0/usr/bin]$
***@***.***:~/telegraf/telegraf-1.26.0/usr/bin]$ sudo netstat
-tupnl
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State
    PID/Program name
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN
     1006/sshd: /usr/sbi
tcp6       0      0 :::9100                 :::*                    LISTEN
     1068/node_exporter
tcp6       0      0 :::8080                 :::*                    LISTEN
     1867/rootlessport
tcp6       0      0 :::22                   :::*                    LISTEN
     1006/sshd: /usr/sbi
tcp6       0      0 :::3000                 :::*                    LISTEN
     1540/grafana-server
udp        0      0 127.0.0.1:323           0.0.0.0:*
    995/chronyd
udp6       0      0 ::1:323                 :::*
     995/chronyd
udp6    2304      0 fe80::20c:29ff:fe7a:546 :::*
     982/NetworkManager
***@***.***:~/telegraf/telegraf-1.26.0/usr/bin]$

**Comment by anpedro** [comment]
figure it out.. fix works, please commit.

**Comment by powersj** [comment]
@anpedro my apologies I looks like I referenced the incorrect PR # in my second message. Thanks for figuring it out and confirming it works. I'll get it reviewed.

**Comment by powersj** [comment]
> I have one other issue related to gnmi and about to open another problem to
look into it.

Please open another issue

**Comment by powersj** [comment]
@anpedro can you do one more round of testing witht he latest artifacts on https://github.com/influxdata/telegraf/pull/12637

We made a change tt the PR and I'd like to ensure it still resolves the issue for you. Thanks!

**Comment by anpedro** [comment]
I need to run this as a podman container, how do I know which telegram
image this fix available?

**Comment by powersj** [comment]
The fix should go into our next bug fix release, v1.25.2 scheduled for next week. It should also land in our nightly images starting tomorrow.
